### ISSUE TITLE
Error when running EmailChecker.valid?

### ISSUE THREAD

**Issue by lowks** [jshmrtn/email_checker#7]
I get this error when using `EmailChecker.valid?`: 

```
     ** (FunctionClauseError) no function clause matching in Kernel.=~/2
     stacktrace:
       (elixir) lib/kernel.ex:1580: Kernel.=~(nil, ~r/^(?<user>[^\s]+)@(?<domain>[^\s]+\.[^\s]+)$/)
       lib/email_checker.ex:28: EmailChecker.valid?/2```

**Comment by rhnonose** [comment]
I'm facing a similar error.

**Comment by maennchen** [comment]
The issue is that `nil` was passed as an argument for `email`. The parameter should be checked in `valid?/2` instead of `Kernel.=~/2`.

**Comment by maennchen** [comment]
Fixed with 3045e12944b19cf362ee4c06ba3fa58eaac95dcc
